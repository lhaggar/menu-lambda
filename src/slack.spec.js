const fixture = require('../fixtures/parsed-menu-content');

describe('src/slack.js', () => {
  describe('buildPayload', () => {
    const setup = ({ day } = {}) => {
      const data = {
        date: new Date('2019-02-01'),
        menuContent: fixture,
        pkgVersion: '1.33.7',
        pkgBugsUrl: 'https://the-bugs-url.com',
        canteenUrl: 'https://the-canteen-url.com',
      };

      const mocks = {
        getDay: jest.fn(() => day),
      };

      jest.doMock('../package.json', () => ({
        version: data.pkgVersion,
        bugs: {
          url: data.pkgBugsUrl,
        },
      }));

      jest.doMock('./config', () => ({
        CANTEEN_URL: data.canteenUrl,
      }));

      jest.doMock('./utils', () => ({
        getDay: mocks.getDay,
      }));

      // eslint-disable-next-line global-require
      const { buildPayload } = require.requireActual('./slack');
      const result = buildPayload(data.date, data.menuContent);

      return {
        data,
        mocks,
        result,
      };
    };

    describe('with canteen and cafe data', () => {
      let result;

      beforeAll(() => {
        ({ result } = setup({ day: 'monday' }));
      });

      afterAll(() => jest.resetModules());

      it('should return a snapshot match', () => {
        expect(result).toMatchSnapshot();
      });
    });

    describe('with no cafe content', () => {
      let result;

      beforeAll(() => {
        ({ result } = setup({ day: 'saturday' }));
      });

      afterAll(() => jest.resetModules());

      it('should return a snapshot match', () => {
        expect(result).toMatchSnapshot();
      });
    });
  });
});
