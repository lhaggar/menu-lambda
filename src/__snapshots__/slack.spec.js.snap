// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/slack.js buildPayload with canteen and cafe data should return a snapshot match 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#C41017",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Genuine chip shop - battered pollock & battered sausages, savaloys
Sides: chips, peas & mushy peas",
      "title": "From the Oven",
    },
    Object {
      "color": "#007404",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Pulled jack fruit
Freshly baked salmon
Served with red onion freekeh, toasted hazelnut and lemon dressing",
      "title": "Healthy Bar",
    },
    Object {
      "color": "#00BFFF",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Chilli glazed chicken, wild rice and red pepper
Sides: glazed sugar snaps and broccoli",
      "title": "Chefs Theatre",
    },
    Object {
      "color": "#e6e600",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Roasted parsnip (v)
Curried cauliflower and bacon",
      "title": "Soup",
    },
    Object {
      "color": "#3126A2",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Breakfast: smoked bacon frittata
Lunch: roasted aubergine and parma ham on focaccia",
      "title": "Caf√©",
    },
    Object {
      "text": "Menu taken from <https://the-canteen-url.com#canteen|Canteen on 14 website>.
Menu Lambda 1.33.7. Any issues or suggestions? <https://the-bugs-url.com|Post on the GitHub!>",
    },
  ],
  "icon_emoji": ":hamburger:",
  "text": "Menu for Friday, February 1, 2019",
  "username": "Canteen on 14",
}
`;

exports[`src/slack.js buildPayload with no cafe content should return a snapshot match 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#C41017",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Genuine chip shop - battered pollock & battered sausages, savaloys
Sides: chips, peas & mushy peas",
      "title": "From the Oven",
    },
    Object {
      "color": "#007404",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Pulled jack fruit
Freshly baked salmon
Served with red onion freekeh, toasted hazelnut and lemon dressing",
      "title": "Healthy Bar",
    },
    Object {
      "color": "#00BFFF",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Chilli glazed chicken, wild rice and red pepper
Sides: glazed sugar snaps and broccoli",
      "title": "Chefs Theatre",
    },
    Object {
      "color": "#e6e600",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "Roasted parsnip (v)
Curried cauliflower and bacon",
      "title": "Soup",
    },
    Object {
      "text": "Menu taken from <https://the-canteen-url.com#canteen|Canteen on 14 website>.
Menu Lambda 1.33.7. Any issues or suggestions? <https://the-bugs-url.com|Post on the GitHub!>",
    },
  ],
  "icon_emoji": ":hamburger:",
  "text": "Menu for Friday, February 1, 2019",
  "username": "Canteen on 14",
}
`;
